ctags.lua

```diff
@@ -220,12 +220,32 @@ local function tselect_cmd(tag, force)
 		vis:info(string.format('Tag not found: %s', tag))
 	else
 		local keys = {}
-		for i = 1, #matches do
-			table.insert(keys, matches[i].desc)
+
+		-- find max width for pretty printing
+		local maxdescwidth = 0
+		for _, match in ipairs(matches) do
+			if match.desc:len() > maxdescwidth then
+				maxdescwidth = match.desc:len()
+			end
+		end
+		local fmt = '%-'..maxdescwidth..'s %s'
+		for _, match in ipairs(matches) do
+			table.insert(keys, fmt:format(match.desc, match.excmd))
+		end
+
+		-- limit vis-menu height to ~1/4 the window height
+		local menu_lines = vis.win.height/4
```

Comment:
We can just write `math.min(vis.win.height/4, #keys+1)`. There is also a potential issue with the division operator, since the result may be a floating point number. As an option `(vis.win.height - vis.win.height%4)/4` can be used to fix it.